### PR TITLE
Support IOBS/SIGIOBS and FOBS/SIGFOBS columns when reading reference intensities from MTZ

### PR DESCRIPTION
**Pull request by rjgildea**
As generated by phenix.cif_as_mtz

### CODE REVIEW

File: newsfragments/2386.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Support ``IOBS/SIGIOBS`` and ``FOBS/SIGFOBS`` columns when reading reference intensities from MTZ files.
```
Comment (graeme-winter) [inline newsfragments/2386.feature:1]: ... in addition to the existing I/SIGI and so on

**Review by graeme-winter — APPROVED** [review]
Change set looks sensible, though the naming of `IOBS` for data computed from a structure strikes me as ironic. I suggested looking at the column types and if this gives a unique solution, use this. That is a suggestion not a requirement.

File: tests/util/test_reference.py [hunk 2]
```diff
@@ -59,3 +62,19 @@ def test_intensities_from_reference_data_file(dials_data):
...
+        output_file_name=os.fspath(mtz_file),
+        wavelength_id=None,
+        crystal_id=None,
+        show_details_if_error=False,
+        output_r_free_label="FREERFLAG",
+    )
+
+    i4 = intensities_from_reference_file(mtz_file)
+    assert i4.data()
+    # In this file, the data is not anomalous
```
[the remark this answers is not in the document]
Comment (rjgildea) [reply tests/util/test_reference.py:79]: Copy-paste from existing comment just above ;)

**Comment by rjgildea** [comment]
> the naming of `IOBS` for data computed from a structure strikes me as ironic.
These are the observed intensities stored in the pdb, e.g. downloaded with `phenix.fetch_pdb --mtz 3I40`:
```
$ phenix.fetch_pdb --mtz 3I40
Model saved to /dls/tmp/rjgildea/3I40.pdb
Sequence saved to /dls/tmp/rjgildea/3I40.fa
Structure factors saved to /dls/tmp/rjgildea/3I40-sf.cif
Model saved to /dls/tmp/rjgildea/3I40.cif
Can't determine output label for r3i40sf,_refln.crystal_id - skipping.
Can't determine output label for r3i40sf,_refln.wavelength_id - skipping.
Can't determine output label for r3i40sf,_refln.scale_group_code - skipping.
fvals [0, 1]
R-free-flags: fraction_free=0.047
R-free-flags: missing 2 reflections
Converted structure factors saved to 3I40.mtz
$ iotbx.mtz.dump 3I40.mtz 
Processing: 3I40.mtz
Title: phenix.cif_as_mtz
Space group symbol from file: I213
Space group number from file: 199
Space group from matrices: I 21 3 (No. 199)
Point group symbol from file: 23
Number of crystals: 2
Number of Miller indices: 6703
Resolution range: 54.6028 1.85014
History:
Crystal 1:
  Name: HKL_base
  Project: HKL_base
  Id: 0
  Unit cell: (77.22, 77.22, 77.22, 90, 90, 90)
  Number of datasets: 1
  Dataset 1:
    Name: HKL_base
    Id: 0
    Wavelength: 0
    Number of columns: 0
Crystal 2:
  Name: crystal_0
  Project: project_0
  Id: 2
  Unit cell: (77.22, 77.22, 77.22, 90, 90, 90)
  Number of datasets: 1
  Dataset 1:
    Name: dataset
    Id: 1
    Wavelength: 0
    Number of columns: 6
    label        #valid  %valid  min     max type
    H              6703 100.00% 0.00   24.00 H: index h,k,l
    K              6703 100.00% 1.00   41.00 H: index h,k,l
    L              6703 100.00% 0.00   41.00 H: index h,k,l
    R-free-flags   6703 100.00% 0.00    1.00 I: integer
    FOBS           6701  99.97% 0.00 1000.00 F: amplitude
    SIGFOBS        6701  99.97% 0.10   14.40 Q: standard deviation
```

**Pull request merged**